### ISSUE TITLE
Hybracter v0.11.0 - Error during LRGE step (Too few reads)

### ISSUE THREAD

**Issue by llk578496** [gbouras13/hybracter#110]
Hi @gbouras13 ,

We encountered an error while running `Hybracter v0.11.0` with this command:

`hybracter long-single -l 02_fastq/BA00000001.fastq.gz -s BA00000001 -c 1900000 -o 03_assembly/03.1_hybracter/BA00000001 -t 64 --subsample_depth 120 --min_depth 100 --medakaModel r1041_e82_400bps_bacterial_methylation --medaka_override`

However, we encountered an error during the lrge step:

```
[2024-12-04T23:43:21Z INFO  lrge] Running two-set strategy with 10000 target reads and 5000 query reads
Error: Failed to generate estimate

Caused by:
    Too few reads requested: Number of reads in FASTQ file (1) is <= query number of reads (5000)
[Thu Dec  5 07:43:21 2024]
Error in rule lrge:
    jobid: 4
    input: 02_fastq/BA00000001.fastq.gz
    output: 03_assembly/03.1_hybracter/BA00000001/processing/chrom_size/BA00000001_lrge_estimated_chromosome_size.txt
    log: 03_assembly/03.1_hybracter/BA00000001/stderr/lrge/BA00000001.log (check log file(s) for error details)
    conda-env: /home/gilmansiu3/miniforge3/envs/hybracter-v0.11.0/lib/python3.12/site-packages/hybracter/workflow/conda/024e350392d5612938570ada6f9b3299_
    shell:
        
        if [[ 02_fastq/BA00000001.fastq.gz =~ \.gz$ ]]; then
        INPUT_CMD="zcat 02_fastq/BA00000001.fastq.gz"
        else
        INPUT_CMD="cat 02_fastq/BA00000001.fastq.gz"
        fi

        # Now run the awk command on the decompressed file
        if $INPUT_CMD | awk '{if (NR % 4 == 1) count++} END {exit count > 10000 ? 0 : 1}'; then
            # There are more than 10,000 reads, run lrge with defaults
            lrge -t 8 -s 42 02_fastq/BA00000001.fastq.gz -o 03_assembly/03.1_hybracter/BA00000001/processing/chrom_size/BA00000001_lrge_estimated_chromosome_size.txt
        else
            # There are less than 10,000 reads, use the all-vs-all strategy with all available reads
            lrge -t 8 -s 42 -n 10000 02_fastq/BA00000001.fastq.gz -o 03_assembly/03.1_hybracter/BA00000001/processing/chrom_size/BA00000001_lrge_estimated_chromosome_size.txt
        fi

        
        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)
Logfile 03_assembly/03.1_hybracter/BA00000001/stderr/lrge/BA00000001.log not found.

Removing output files of failed job lrge since they might be corrupted:
03_assembly/03.1_hybracter/BA00000001/processing/chrom_size/BA00000001_lrge_estimated_chromosome_size.txt
Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
Complete log: .snakemake/log/2024-12-05T074308.272662.snakemake.log
WorkflowError:
At least one job did not complete successfully.
[2024:12:05 07:43:21] ERROR: Snakemake failed

```

We have also attached the log file for your reference: 
[hybracter.log](https://github.com/user-attachments/files/18015453/hybracter.log)

This sample processed successfully in `v0.10.1`, suggesting the input file is valid. Could you help us resolve this issue?

Thank you very much!

Best regards,
Eddie

**Comment by gbouras13** [comment]
Hi @llk578496 - apologies for this, the v0.11.0 clearly has some bugs.

Do you know how many reads the input FASTQ has (clearly not 1!)? And would you be able to share the FASTQ (or a subset of it if it also breaks hybracter) with me and/or @mbhall88 so we can reproduce the error? george.bouras@adelaide.edu.au is my email too.

George

**Comment by mbhall88** [comment]
Yes if you could somehow share the FASTQ file so we can debug this it would be most appreciated

**Comment by llk578496** [comment]
Hi @gbouras13 and @mbhall88 ,

Here are the statistics for the FASTQ file:
[BA00000001.stats.txt](https://github.com/user-attachments/files/18017266/BA00000001.stats.txt)

I've also shared the FASTQ file with George's email.
Thank you very much!

Best regards,
Eddie

**Comment by llk578496** [comment]
Dear George and Michael,

Attached is the OneDrive link for the FASTQ file.
[https://res.cdn.office.net/assets/mail/file-icon/png/folder_16x16.png]#110<https://connectpolyu-my.sharepoint.com/:f:/g/personal/20113427r_connect_polyu_hk/ElzNkuE5wDlGg0sRxtqb5qYBcfwUDgyzHzZXVOOSMuzxdg?e=eVXdEX>

Thank you very much!

Best regards,
Lee Lam Kwong Eddie
________________________________

**Comment by gbouras13** [comment]
Hi @llk578496 ,

If you urgently need to run this, I believe the error is to do with the way `lrge` parsed the compression of the FASTQ file. 

Therefore, try unzipping and gzipping the file - this worked for me

e.g.

```
gunzip BA00000001.fastq.gz
gzip BA00000001.fastq
lrge BA00000001.fastq.gz 
[2024-12-05T05:16:51Z INFO  lrge] Running two-set strategy with 10000 target reads and 5000 query reads
[2024-12-05T05:17:37Z WARN  liblrge::twoset] 338 (6.76%) query read(s) did not overlap any target reads
[2024-12-05T05:17:37Z INFO  lrge] Estimated genome size: 1.95 Mbp (IQR: 1.58 Mbp - 2.10 Mbp)
1950278
[2024-12-05T05:17:37Z INFO  lrge] Done!
```

Michael has identified the error and will work on a fix, so if it isn't urgent, feel free to wait. Thank you so much for picking up the error. 

George

**Comment by llk578496** [comment]
Hi @gbouras13 and @mbhall88,

Thank you for your help! The file compression fix worked successfully. We'll wait for the new update.

Best regards,
Eddie

**Comment by mbhall88** [comment]
I have fixed lrge in v0.1.3 which is now up on conda. So if you remove the lrge conda env that hybracter created and rerun, it should use that latest version. 
thank you very much for reporting the bug

**Comment by gbouras13** [comment]
Closing this issue now as `lrge` seems to have solved the issue
